packages/number-field/src/vaadin-number-field.js

```diff
@@ -467,9 +492,14 @@ export class NumberField extends InputFieldMixin(SlotStylesMixin(ThemableMixin(E
   checkValidity() {
     if (
       this.inputElement &&
-      (this.required || this.min !== undefined || this.max !== undefined || this.__validateByStep)
+      (this.required || this.min !== undefined || this.max !== undefined || this.step !== undefined)
     ) {
-      return this.inputElement.checkValidity();
+      if (this.value == null || this.value == '') {
```

Comment:
Seems I can programmatically set a non-numeric value for the field as long as the value starts with a number:
![Screenshot 2021-11-23 at 12 17 07](https://user-images.githubusercontent.com/1222264/143006785-c62e5179-b2b6-4aa7-8e78-e6ffd0428e04.png)